File: src/traust_engine/data/secure-code-audit/opengrep-rules/typescript/ssrf.ts
import axios from "axios";

export async function proxy(req: any, res: any): Promise<void> {
  const target = req.query.endpoint;
  // ruleid: traust-ts-ssrf-request-taint
  await fetch(target);

  // ruleid: traust-ts-ssrf-request-taint
  await axios.get(`https://${req.params.cluster}/api/v1/status`);

  // ok: traust-ts-ssrf-request-taint
  await fetch("https://api.openshift.com/healthz");
}

File: src/traust_engine/data/secure-code-audit/opengrep-rules/python/ssrf.py
import requests
from flask import request


def proxy():
    target = request.args.get("url")
    # ruleid: traust-python-ssrf-request-taint
    requests.get(target)

    # ruleid: traust-python-ssrf-request-taint
    requests.post("https://" + request.form.get("backend") + "/api")

    # ok: traust-python-ssrf-request-taint
    requests.get("https://api.openshift.com/healthz")
